components/renku_data_services/base_models/core.py

```diff
@@ -207,6 +207,62 @@ def __repr__(self) -> str:
         return self.value
 
 
+@dataclass(frozen=True, eq=True)
+class EntityPath:
+    """The collection of slugs that makes up the path to an entity in Renku."""
+
+    slugs: list[Slug] = field(default_factory=list)
+
+    def __repr__(self) -> str:
+        return "/".join([slug.value for slug in self.slugs])
+
+    def __truediv__(self, other: Slug | str | Self) -> "EntityPath":
+        """Create new entity path with an extra slug."""
+        slugs = [slug for slug in self.slugs]
+        if isinstance(other, Slug):
+            slugs.append(other)
+        elif isinstance(other, str):
+            slugs.append(Slug(other))
+        elif type(self) is type(other):
+            slugs.extend(other.slugs)
+        else:
+            raise errors.ValidationError(
+                message="A path can be constructed from an entity path and a slug,string or entity path, "
+                f"but the 'divisor' is of type {type(other)}"
+            )
+        return EntityPath(slugs)
+
+    @classmethod
+    def join(cls, *slugs: Slug | str) -> Self:
```

Comment:
suggestion: I think truediv can be expressed using join or the other way around, whichever is easier, to reduce code duplication.

Reply:
Done